perfkitbenchmarker/linux_packages/blaze.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2014 PerfKitBenchmarker Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Module containing blaze installation and cleanup functions."""
+
+import os
+from perfkitbenchmarker import vm_util
+
+BLAZE_VERSION = '3.0'
+BLAZE_TAR = 'blaze-%s.tar.gz' % BLAZE_VERSION
+BLAZE_DIR = '%s/blaze-%s' % (vm_util.VM_TMP_DIR, BLAZE_VERSION)
+BLAZE_TAR_URL = (
+    'https://bitbucket.org/blaze-lib/blaze/downloads/%s' % BLAZE_TAR)
+CONFIG = 'Configfile_blaze'
+MAX_BLAZE_CACHE_SIZE_IN_B = 100000000
+
+
+def _Configure(vm):
+  """Configure and build blaze library.
+
+  See https://bitbucket.org/blaze-lib/blaze/wiki/Configuration%20Files
+  for more details.
+  """
+  vm.PushDataFile(CONFIG, BLAZE_DIR)
+  # Adjust cache size
+  cache_in_KB, _ = vm.RemoteCommand(
+      'cat /proc/cpuinfo | grep "cache size" | awk \'{print $4}\'')
+  cache_in_B = int(1024 * sum([float(c) for c in cache_in_KB.split()]))
+  vm.RemoteCommand(
+      'sed -i \'s/constexpr size_t cacheSize = 3145728UL;/constexpr '
+      'size_t cacheSize = %sUL;/g\' %s' % (
+          max(cache_in_B, MAX_BLAZE_CACHE_SIZE_IN_B - 1), os.path.join(
+              BLAZE_DIR, 'blaze', 'config', 'CacheSize.h')))
+  vm.RemoteCommand('cd %s; ./configure %s; make' % (BLAZE_DIR, CONFIG))
```

Comment:
nit: I think chaining commands with && leads to easier to read error messages because it won't continue executing if one of the earlier steps fails